app/controllers/charityInformation/CanWeSendToThisAddressController.scala

```diff
@@ -0,0 +1,99 @@
+import play.api.mvc._
+import repositories.UserAnswerRepository
+import views.html.charityInformation.CanWeSendToThisAddressView
+
+import scala.concurrent.Future
+
+class CanWeSendToThisAddressController  @Inject()(sessionRepository: UserAnswerRepository,
+                                                  navigator: CharityInformationNavigator,
+                                                  identify: AuthIdentifierAction,
+                                                  getData: UserDataRetrievalAction,
+                                                  requireData: DataRequiredAction,
+                                                  formProvider: CanWeSendToThisAddressFormProvider,
+                                                  val controllerComponents: MessagesControllerComponents,
+                                                  view: CanWeSendToThisAddressView
+                                                 )(implicit appConfig: FrontendAppConfig) extends LocalBaseController {
+
+  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+
+      val form = formProvider()
+
+      val preparedForm = request.userAnswers.get(CanWeSendToThisAddressPage) match {
+        case None => form
+        case Some(value) => form.fill(value)
+      }
+
+      getCharityInformationAddressLookup {
+        charityInformationAddressLookup =>
+
+          lazy val addressList = charityInformationAddressLookup
+
+
+      Future.successful(Ok(view(preparedForm, mode, addressList)))
+      }
+  }
+
+  def onSubmit(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+
+      val form = formProvider()
+
+      form.bindFromRequest().fold(
+        formWithErrors =>
+          getCharityInformationAddressLookup {
+            charityInformationAddressLookup =>
+          Future.successful(BadRequest(view(formWithErrors, mode, charityInformationAddressLookup)))
+          },
+
+        value =>
+          for {
+            updatedAnswers <- Future.fromTry(request.userAnswers.set(CanWeSendToThisAddressPage, value).flatMap(_.set(Section1Page, false)))
+            _ <- sessionRepository.set(updatedAnswers)
+          } yield Redirect(navigator.nextPage(CanWeSendToThisAddressPage, mode, updatedAnswers))
+      )
+  }
+
+    private def getCharityInformationAddressLookup(block: String => Future[Result])
+                                          (implicit request: DataRequest[AnyContent]): Future[Result] = {
+
+    request.userAnswers.get(CharityInformationAddressLookupPage).map {
+      charityInformationAddressLookup =>
+
+        lazy val addressList = charityInformationAddressLookup.lines.mkString("",", ",", ")
+        lazy val postcode = charityInformationAddressLookup.postcode.mkString("")
+        lazy val country = charityInformationAddressLookup.country.name.mkString(" ", "", "")
```

Comment:
can we make this line more readable, may be something like this?
`val addressList = charityInformationAddressLookup.lines
val postcode = charityInformationAddressLookup.postcode.fold(Seq[String]())(Seq(_))
val country = Seq(charityInformationAddressLookup.country.name)
block(Seq(addressList, postcode, country).flatten.mkString(", "))`